include/seqan3/alignment/configuration/align_config_alignment_result_capture.hpp

```diff
@@ -0,0 +1,78 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::align_cfg::alignment_result_capture.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/configuration/detail.hpp>
+#include <seqan3/alignment/pairwise/alignment_result.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+#include <seqan3/std/type_traits>
+
+namespace seqan3::detail
+{
+
+/*!\brief The configuration element storing the captured alignment result type.
+ * \ingroup alignment_configuration
+ *
+ * \tparam alignment_result_t The type of the alignment result to capture; must be a type specialisation of
+ *                            seqan3::alignment_result.
+ *
+ * \details
+ *
+ * Implementation of the alignment result capture config.
+ *
+ * \see seqan3::align_cfg::alignment_result_capture
+ */
+template <typename alignment_result_t>
+//!\cond
+    requires is_type_specialisation_of_v<alignment_result_t, alignment_result>
+//!\endcond
+struct alignment_result_capture_element :
+    public pipeable_config_element<alignment_result_capture_element<alignment_result_t>,
+                                   std::type_identity<alignment_result_t>>
+{
+    //!\brief Internal id to check for consistent configuration settings.
+    static constexpr detail::align_config_id id{detail::align_config_id::alignment_result_capture};
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::align_cfg
+{
+/*!\if DEV
```

Comment:
is \if the same as \cond?

Reply:
They are somewhat similar. The problem is that the function itself has a `\cond \endcond` expression around the reuires expression. But I had some weird issues with the nested `cond` expression (due to the requires block) so that was an elegant solution to let them only appear in dev docu.